### PR TITLE
Add and use new `IconButton` component

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR takes care of tasks pertaining to several goals. 

## User-visible Changes

* The top-bar icons are now the same color as the annotation-card action icons (H staff may review [this Slack thread](https://hypothes-is.slack.com/archives/C07NXBDNW/p1576246965000500) )
* The top-bar icons are now all SVGs. The following icons are updated in this PR:
    * search
    * sort
    * user/profile

Before:

image

After (EDITED to show updated sort icon):

image

(NB: The icons are still the same size as before; my slightly imprecise screen grabs are scaled here differently and make it appear that there was a size change—there was not).

Other underlying changes here do not result in user-visible effects.

## Other Changes

A new component, `IconButton` has been introduced, which is then used in several places where the client wants to render a button that only has an icon on it. In this PR, `IconButton` is used in these components:

* `TopBar`
* `AnnotationActionBar`
* `AnnotationShareControl` (renders inside of `AnnotationActionBar`)
* `SearchInput` (renders inside of `TopBar`)
* `ShareAnnotationsPanel` (renders inside of `TopBar`)
* `SortMenu` (renders inside of `TopBar`)

The result is that all buttons rendered in the app's `TopBar` and in the annotation's `ActionBar` use `IconButton`.

Part of https://github.com/hypothesis/client/issues/1561

## Next Steps

* Convert icons in `MarkdownEditor` to use `IconButton` (the diff for this PR was getting chattery, so I thought I'd wait on this)
* Extract `ActionButton` component (already done on local branch)
* Rework the way we use `primary-action-btn` to be consistent with other preact components

### CODE REVIEW

File: src/styles/sidebar/components/icon-button.scss [hunk 1]
```diff
@@ -1,7 +1,7 @@
 @use "../../mixins/buttons";
 @use "../../variables" as var;
 
-.annotation-action-button {
+.icon-button {
   @include buttons.button-base;
 
   &.is-active {
```
Comment (LMS007) [inline src/styles/sidebar/components/icon-button.scss]: Wondering if base-button should get a default `is-active` styling? like grey-6 or something

File: src/sidebar/components/icon-button.js [hunk 2]
```diff
@@ -7,37 +7,41 @@ const { createElement } = require('preact');
...
+  /** Optional additional class(es) to apply to the component element */
+  className: propTypes.string,
   /** The name of the SVGIcon to render */
   icon: propTypes.string.isRequired,
   /** Is this button currently in an "active" or "on" state? */
   isActive: propTypes.bool,
-  /** a label used for the `title` and `aria-label` attributes */
-  label: propTypes.string.isRequired,
+  /** a value used for the `title` and `aria-label` attributes */
+  title: propTypes.string.isRequired,
```
Comment (LMS007) [inline src/sidebar/components/icon-button.js]: we care about a-z ordering here?

File: src/sidebar/components/icon-button.js [hunk 3]
```diff
@@ -7,37 +7,41 @@ const { createElement } = require('preact');
...
   icon,
   isActive = false,
-  label,
+  title,
   onClick = () => null,
 }) {
   return (
     <button
-      className={classnames('annotation-action-button', {
+      className={classnames('icon-button', className, {
```
[the remark this answers is not in the document]
Comment (robertknight) [reply src/sidebar/components/icon-button.js]: 5. If you can make use of CSS variables (so, everywhere except IE 11), then you can let the external class set a variable which the internal class makes use of.
6. Decide f**k it, and use `!important` on the external class 😬 

I'm sure there are other approaches I've missed here.

File: src/sidebar/components/top-bar.js [hunk 4]
```diff
@@ -13,48 +13,12 @@ const { withServices } = require('../util/service-context');
 const uiConstants = require('../ui-constants');
 
 const GroupList = require('./group-list');
+const IconButton = require('./icon-button');
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/sidebar/components/top-bar.js]: 100% agree! I have on my list already to consider suggesting that the up/down buttons in the sort icon be pulled closer to each other in the source SVG, as it does _seem_ like the spacing here is tighter or uneven.

**Comment by lyzadanger** [comment]
Non-negligible changes since last review:

* `sort` icon is updated to have tighter up-down arrows
* Small spacing oddity fixed in `search-input` (I'll comment with details)
* `useCompactStyle` prop added to `IconButton`, mirroring how `ActionButton` handles this situation currently
* Component handles its own padding, but still allows arbitrary `className` as there are several needs for this across components. e.g. `SearchInput` adds tab-ordering; `TopBar` needs a different active-state color (I do hope we'll be able to consolidate that latter difference at some point). I spent a while thinking this one through; I think this hybrid solution reflects the most straightforward way to tackle this for the present. I'll continue to consider this as I move through all of this refactoring.

File: src/styles/sidebar/components/search-input.scss [hunk 5]
```diff
@@ -27,11 +27,8 @@
   width: 100%;
 
   // The search box expands when focused, via a change in the
-  // `max-width` property. In Safari, the <input> will not accept
```
Comment (lyzadanger) [inline src/styles/sidebar/components/search-input.scss]: I read up on the original ticket, which is four years old. _It does not appear that this bug is still manifest in Safari_. This `0.1px` rule caused occasional layout issues in Chrome in certain circumstances, likely due to pixel rounding. You can see it here—note the one-pixel gap between the search icon and the one next to it:

Screen Shot 2019-12-17 at 9 39 56 AM

**Pull request merged**